lib/SimpleSAML/Database.php

```diff
@@ -261,7 +255,7 @@ public function write($stmt, $params = [])
 
         if (is_array($params)) {
             $obj = $this->query($db, $stmt, $params);
-            return ($obj === false) ? $obj : $obj->rowCount();
+            return $obj->rowCount();
```

Comment:
Since `query()` now is guaranteed to return a `PDOStatement`, I think we can conflate these two statements:

```php
            return $this->query($db, $stmt, $params)->rowCount();
```

Reply:
Check